dao/mongodb.rya/src/main/java/org/apache/rya/mongodb/aggregation/AggregationPipelineQueryNode.java

```diff
@@ -0,0 +1,882 @@
+            p = RdfToRyaConversions.convertURI((URI) predVar.getValue());
+        }
+        if (objVar != null && objVar.getValue() != null) {
+            o = RdfToRyaConversions.convertValue(objVar.getValue());
+        }
+        if (contextVar != null && contextVar.getValue() instanceof URI) {
+            c = RdfToRyaConversions.convertURI((URI) contextVar.getValue());
+        }
+        RyaStatement rs = new RyaStatement(s, p, o, c);
+        DBObject obj = strategy.getQuery(rs);
+        // Add path prefix, if given
+        if (path.length > 0) {
+            StringBuilder sb = new StringBuilder();
+            for (String str : path) {
+                sb.append(str).append(".");
+            }
+            String prefix = sb.toString();
+            Set<String> originalKeys = new HashSet<>(obj.keySet());
+            originalKeys.forEach(key -> {
+                Object value = obj.removeField(key);
+                obj.put(prefix + key, value);
+            });
+        }
+        return (BasicDBObject) obj;
+    }
+
+    private static String valueFieldExpr(String varName) {
+        return "$" + VALUES + "." + varName;
+    }
+    private static String hashFieldExpr(String varName) {
+        return "$" + HASHES + "." + varName;
+    }
+    private static String typeFieldExpr(String varName) {
+        return "$" + TYPES + "." + varName;
+    }
+    private static String joinFieldExpr(String triplePart) {
+        return "$" + JOINED_TRIPLE + "." + triplePart;
+    }
+
+    /**
+     * Get an object representing the value field of some value expression, or
+     * return null if the expression isn't supported.
+     */
+    private Object valueFieldExpr(ValueExpr expr) {
+        if (expr instanceof Var) {
+            return valueFieldExpr(((Var) expr).getName());
+        }
+        else if (expr instanceof ValueConstant) {
+            return new Document("$literal", ((ValueConstant) expr).getValue().stringValue());
+        }
+        else {
+            return null;
+        }
+    }
+
+    private final List<Bson> pipeline;
+    private final MongoCollection<Document> collection;
+    private final Set<String> assuredBindingNames;
+    private final Set<String> bindingNames;
+    private final BiMap<String, String> varToOriginalName;
+
+    private String replace(String original) {
+        if (varToOriginalName.containsValue(original)) {
+            return varToOriginalName.inverse().get(original);
+        }
+        else {
+            String replacement = "field-" + UUID.randomUUID();
+            varToOriginalName.put(replacement, original);
+            return replacement;
+        }
+    }
+
+    /**
+     * Create a pipeline based on a StatementPattern.
+     * @param baseSP The leaf node in the query tree.
+     */
+    public AggregationPipelineQueryNode(MongoCollection<Document> collection, StatementPattern baseSP) {
+        Preconditions.checkNotNull(collection);
```

Comment:
The checks return the checked object, so this could be 
```
import static java.util.Objects.requireNonNull;
...
this.collection = requireNonNull(collection)
```
Comment applies through out this review.

Reply:
Changed.